### PR TITLE
`treehouses services piwigo` (fixes #1421)

### LINKED ISSUE #1421: `treehouses services piwigo`?
https://hub.docker.com/r/linuxserver/piwigo

### PR DESCRIPTION
**Pull request by hiroTochigi**
fixes #1421

Database address is ip address `192.168.0.*` ~~or onion address.~~
Database user is `root`
Database password is `my-secret-pw`
Database name is `piwigo`

Administration section is up to you.

After you install it through the local connection, you can access it via onion address.

### CODE REVIEW

**Comment by kjong** [comment]
```
root@kjong:~/cli# ./cli.sh services piwigo install
WARNING: The MYSQL_ROOT_PASSWORD_VAR variable is not set. Defaulting to a blank string.
WARNING: The MYSQL_DATABASE_VAR variable is not set. Defaulting to a blank string.
Pulling db     ... done
```

File: services/install-piwigo.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+fi
+EOF
+}
+
+# environment var
+function uses_env {
+  echo "MYSQL_ROOT_PASSWORD_VAR=my-secret-pw"
+  echo "MYSQL_DATABASE_VAR=piwigo" 
+} > /srv/piwigo/.env
+
```
Comment (kjong) [inline services/install-piwigo.sh:51]: The actual .env file creation should be inside `install` and `uses_env` should only `echo true` or `echo false`

**Comment by hiroTochigi** [comment]
```
root@treehouses:~/cli/tests# bats services/piwigo.bats 
 ✓  services piwigo info
 ✓  services piwigo install
 ✓  services piwigo up
 ✓  services piwigo start
 ✓  services piwigo restart
 ✓  services available
 ✓  services running
 ✓  services running full
 ✓  services installed
 ✓  services installed full
 ✓  services ports
 ✓  services piwigo port
 ✓  services piwigo ps
 ✓  services piwigo url
 ✓  services piwigo autorun
 ✓  services piwigo autorun true
 ✓  services piwigo stop
 ✓  services piwigo down
 ✓  services piwigo icon
 ✓  services piwigo cleanup

20 tests, 0 failures
```

File: services/install-piwigo.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    volumes:
+      - /srv/piwigo/datadir:/var/lib/mysql
+EOF
+
+  # create .env with default values
+  {
+    echo "MYSQL_ROOT_PASSWORD_VAR=my-secret-pw"
+    echo "MYSQL_DATABASE_VAR=piwigo" 
+  } > /srv/piwigo/.env
+
```
Comment (kjong) [inline services/install-piwigo.sh:41]: Could you switch this to using a heredoc instead to keep it consistent?

File: services/install-piwigo.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  # add autorun
+  cat << EOF > /srv/piwigo/autorun
+piwigo_autorun=true
+
+if [ "$piwigo_autorun" = true ]; then
+  treehouses services piwigo up
+fi
+EOF
```
Comment (kjong) [inline services/install-piwigo.sh:50]: You could add two extra newlines between `fi` and `EOF`.
Not functionally important, but helps preserve the formatting in `/boot/autorun`

**Pull request merged**